lib/absinthe/plug/graphiql.ex

```
@@ -221,6 +221,20 @@ defmodule Absinthe.Plug.GraphiQL do
           raise "invalid default url: #{inspect val}"
        end
 
+     config = case config[:socket_url] do
+        nil -> config
+        val when is_binary(val) -> Map.put(config, :socket_url, val)
+        {module, fun} when is_atom(fun) ->
+          url =
+            module
+            |> function_exported?(fun, 1)
```

I would prefer if we raised here instead of just ignoring it in the `call_exported_function` function.

---

sure! Do you want me to do that in `call_exported_function`? so that its the same for `default_headers`, `default_url` and `socket_url`?

---

@benwilson512 I've updated! its a bit messy, since both 1 or 0 arity is supported, and for the socket stuff, there is also the `socket` config option